File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKActionRemoveFromParent.java
package br.com.insanitech.spritekit;

/**
 * Created by anderson on 06/01/17.
 */

class SKActionRemoveFromParent extends SKAction {
    @Override
    void computeStart() {

    }

    @Override
    void computeAction(long elapsed) {
        getParent().removeFromParent();
    }

    @Override
    void computeFinish() {

    }

    @Override
    boolean willHandleFinish() {
        return false;
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKActionRemoveFromParent.kt
package br.com.insanitech.spritekit

/**
 * Created by anderson on 06/01/17.
 */

internal class SKActionRemoveFromParent : SKAction() {
    internal override fun computeStart() {

    }

    internal override fun computeAction(elapsed: Long) {
        parent?.removeFromParent()
    }

    internal override fun computeFinish() {

    }
}
